lnwallet/commitment.go

```diff
@@ -1017,9 +1025,24 @@ func (cb *CommitmentBuilder) createUnsignedCommitmentTx(ourBalance,
 	}
 
 	// Sort the transactions according to the agreed upon canonical
-	// ordering. This lets us skip sending the entire transaction over,
-	// instead we'll just send signatures.
-	InPlaceCommitSort(commitTx, cltvs)
+	// ordering (which might be customized for custom channel types, but
+	// deterministic and both parties will arrive at the same result). This
+	// lets us skip sending the entire transaction over, instead we'll just
+	// send signatures.
+	if auxLeaves.IsSome() &&
+		auxLeaves.UnsafeFromSome().CustomOutputOrder != nil {
```

Comment:
If we want to mask this a bit more, then we can have a functional option to pass in a new custom sorting interface/type. Default is the existing one, then we know to override it at the config level for these special channels.

Reply:
Good point. I'm meeting you half way: Instead of passing _in_ a function, we now require a sort function to be set if there are aux leaves. But that way at least the sorting logic is not in the `lnd` code base anymore.